### PR TITLE
Add flake8 check for line length in docstrings

### PR DESCRIPTION
**Pull request by namurphy**
This PR will add a check in flake8 to make sure that the lengths of docstrings don't exceed some limit.  The PEP 8 recommendation ≤ 72 characters, but since we have a lot of docstrings longer than that, I'm going to set the max doc length on the high end of that, rather than attempt to fix everything.  Ideally, this would end up being able to be done using a docstring autoformatter in the future, though some improvements to the autoformatters will be necessary before we can do that in maybe ∼1–3 years.  It should show up as error code `W505` from pycodestyle.

### CODE REVIEW

**Comment by namurphy** [comment]
In addition to the "procrastination-driven development" label, I wonder if we should have a "rainy day PR" label for stuff you can work on in the delay between moves when playing Civilization IV...

File: plasmapy/analysis/nullpoint.py [hunk 1]
```diff
@@ -164,66 +166,64 @@ def _vector_space(
...
-        If not given, then range values are used to construct a
-        uniform array on that interval.
+        The array representing the coordinates in the z-dimension. If
+        not given, then range values are used to construct a uniform
+        array on that interval.
 
-    x_range : |array_like|
-        A 1 by 2 array containing the range of x-values for the vector spaces.
-        If not given, the default interval [0,1] is assumed.
+    x_range : |array_like|, default: ``[0, 1]``
```
Comment (namurphy) [inline plasmapy/analysis/nullpoint.py:184]: Using a convention from matplotlib's dev guide, which is also in #1757.

File: plasmapy/diagnostics/langmuir.py [hunk 2]
```diff
@@ -638,7 +642,8 @@ def get_ion_density_LM(
         The electron temperature in units convertible to eV.
 
     probe_area : `~astropy.units.Quantity`
-        The area of the probe exposed to plasma in units convertible to m^2.
+        The area of the probe exposed to plasma in units convertible to
+        m\ :math:`2`\ .
```
Comment (namurphy) [inline plasmapy/diagnostics/langmuir.py:646]:
```suggestion
        m\ :sup:`2`\ .
```

File: plasmapy/diagnostics/langmuir.py [hunk 3]
```diff
@@ -700,12 +704,14 @@ def get_electron_density_LM(
...
```
Comment (namurphy) [inline plasmapy/diagnostics/langmuir.py:708]:
```suggestion
        m\ :sup:`2`\ .
```

File: plasmapy/diagnostics/langmuir.py [hunk 4]
```diff
@@ -1159,7 +1165,8 @@ def get_ion_density_OML(
         The swept probe characteristic that is to be analyzed.
 
     probe_area : `~astropy.units.Quantity`
-        The area of the probe exposed to plasma in units convertible to m^2.
+        The area of the probe exposed to plasma in units convertible to
+        m\ :math:`2`\ .
```
Comment (namurphy) [inline plasmapy/diagnostics/langmuir.py:1169]:
```suggestion
        m\ :sup:`2`\ .
```

File: plasmapy/particles/ionization_state.py [hunk 5]
```diff
@@ -389,9 +389,9 @@ def __eq__(self, other):
 
         Examples
         --------
-        >>> IonizationState('H', [1, 0], tol=1e-6) == IonizationState('H', [1, 1e-6], tol=1e-6)
+        >>> IonizationState('H', [1, 0], tol=1e-6) == IonizationState('H', [1, 1e-6], tol=1e-6)  # noqa: W505
```
Comment (namurphy) [inline plasmapy/particles/ionization_state.py:392]: `W505` is the pycodestyle (via flake8) error code for maximum docstring line length.

**Comment by namurphy** [comment]
This ended up being a pretty long pull request, but nearly all of the changes are for line length, with only a few minor reST fixes plus the flake8 configuration stuff.  I went through the changes twice, so it should be fine to skim them.

**Review by StanczakDominik — APPROVED** [review]
Awesome, happy to have it go in; ideally after #1758, though. That one moves flake8 settings to tox.ini.

**Comment by namurphy** [comment]
> Awesome, happy to have it go in; ideally after #1758, though. That one moves flake8 settings to tox.ini.

Agreed, getting #1758 merged first sounds like better option.

**Comment by StanczakDominik** [comment]
We removed this  in [`0d30858` (#1828)](https://github.com/PlasmaPy/PlasmaPy/pull/1828/commits/0d30858f1c9fd195c4510a5f41d5de212c1ef5b6), temporarily. We'd really like a better linter, but for now it's just getting in the way, mostly.

**Pull request merged**